src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -105,14 +105,17 @@ def subject(self):
         self._backend.openssl_assert(subject != self._backend._ffi.NULL)
         return _decode_x509_name(self._backend, subject)
 
-    @property
-    def signature_hash_algorithm(self):
+    def _get_signature_algorithm_oid(self):
         alg = self._backend._ffi.new("X509_ALGOR **")
         self._backend._lib.X509_get0_signature(
             self._backend._ffi.NULL, alg, self._x509
         )
         self._backend.openssl_assert(alg[0] != self._backend._ffi.NULL)
-        oid = _obj2txt(self._backend, alg[0].algorithm)
+        return _obj2txt(self._backend, alg[0].algorithm)
+
+    @property
+    def signature_hash_algorithm(self):
+        oid = self._get_signature_algorithm_oid()
         try:
             return x509._SIG_OIDS_TO_HASH[oid]
```

Comment:
I'm always in favor of simplifying implementations.

On Tue, Aug 30, 2016 at 9:08 AM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/x509.py
> <https://github.com/pyca/cryptography/pull/3124#discussion_r76790437>:
>
> >          alg = self._backend._ffi.new("X509_ALGOR **")
> >          self._backend._lib.X509_get0_signature(
> >              self._backend._ffi.NULL, alg, self._x509
> >          )
> >          self._backend.openssl_assert(alg[0] != self._backend._ffi.NULL)
> > -        oid = _obj2txt(self._backend, alg[0].algorithm)
> > +        return _obj2txt(self._backend, alg[0].algorithm)
> > +
> > +    @property
> > +    def signature_hash_algorithm(self):
> > +        oid = self._get_signature_algorithm_oid()
> >          try:
> >              return x509._SIG_OIDS_TO_HASH[oid]
>
> We've implemented __hash__ on ObjectIdentifier, so yeah we could do that
> and then simplify the implementation a bit.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pyca/cryptography/pull/3124/files/c3d9dde765fccf8bac5e20bd60470ad5fe0c5f11#r76790437>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAADBHGRl_RlMqyC6SFdFAZeYv8O_5wOks5qlCtrgaJpZM4JwfKU>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6